### DISCUSSION TITLE
search all  explore for a title

### DISCUSSION THREAD

**Discussion by blueray453** [papis/papis#872] [Q&A] [answered]
First of all, how to list all explores.

The documentation https://papis.readthedocs.io/en/latest/commands/list.html says I should be able to list explorers

```
$ papis --version
papis, version 0.13

$ papis list --explorers
Usage: papis list [OPTIONS] [QUERY]

Error: No such option: --explorers Did you mean --reverse?
```
But it says No such option.

Is it a bug or am i doing something wrong. `papis list --exporters` also does not work. Same error.

My main issues are, 

- Is there any way I can search all explore for a Title (book / journal article) (exact match).
- Is there any way I can search all explore for a ISBN/DOI (exact match).

**Comment by alexfikl** [comment]
> Is it a bug or am i doing something wrong. papis list --exporters also does not work. Same error.

This was added in the development version that's still unreleased. It's not available on the 0.13 version that I assume you're using, sorry for that! You can try installing the latest version from `git`, but it's a bit of a pain depending on your setup :(

You should be able to see all explorers also listed in the help message (under Commands)
```
papis explore --help
```

> * Is there any way I can search all explore for a Title (book / journal article) (exact match).
> * Is there any way I can search all explore for a ISBN/DOI (exact match).

I'm afraid that's not possible at the moment. It's also quite difficult to pull off, since the importers are pretty different in what type of queries they support.

If you need to import a document using an exact match for the ISBN or DOI, I recommend directly using the `papis add` command. For example, something like
```
papis add --from doi '<SOME_DOI>' <SOME_PDF_MAYBE.pdf>
papis add --from isbn '<SOME_ISBN>' <SOME_PDF_MAYBE.pdf>
```
should only import the documents if an exact match is found. Would that work?

**Reply by blueray453** [reply]
I actually do not want to add. Only view the meta data using json. Like

    $ papis explore isbn -q "Antifragile: Things That Gain from Disorder" export --format json | jq

It there any way I can do it for pdf or epub?

**Reply by alexfikl** [reply, marked as answer]
> I actually do not want to add. Only view the meta data using json

Hm, for DOIs, you can use crossref like this
```
papis explore crossref --filter doi "SOME_DOI" export --format json | jq
```
Does that work? I think that also does an exact match, but I'm not sure.

> It there any way I can do it for pdf or epub?

What do you mean? Find the metadata for a given PDF file?

**Reply by blueray453** [reply]
Yes. Thank you very much.
